index.js

```diff
@@ -543,147 +543,158 @@ async function html(urls, options) {
  */
 
 async function epubgen(data, output_path, options) {
-	const template_base = path.join(__dirname, 'templates/epub/');
-
-	const output = _fs.createWriteStream(output_path);
-	const archive = archiver('zip', {
-		store: true
-	});
-
-	output
-		.on('close', () => {
-			out.write(archive.pointer() + ' total bytes');
-			out.write(
-				'archiver has been finalized and the output file descriptor has closed.'
-			);
-		})
-		.on('end', () => {
-			out.write('Data has been drained');
+	const wrapAsync = inner => {
+		return new Promise((resolve, reject) => {
+			inner(resolve, reject).catch(reject);
 		});
+	};
 
-	archive
-		.on('warning', err => {
-			throw err;
-		})
-		.on('error', err => {
-			throw err;
-		})
-		.pipe(output);
+	return wrapAsync(async resolve => {
```

Comment:
Here I'm wrapping the original code. I'm not using the `reject` function here, because the `wrapAsync` function is catching exception from this code, and rejecting the promise if any exception happens.